### ISSUE TITLE
WP-Maintenance Apache+Nginx+Varnsh

### ISSUE THREAD

**Issue by uhlhosting** [Designmodocom/WP-Maintenance-Mode#139]
It seems after running varnish, my site over maintenance on is not avalable anymore. Alto I am logged in as Admin this is not avaliable anymore. varnish wont cache logged in users, yet still even clearing varnish cache does not help. Any workaround this?

**Comment by georgejipa** [comment]
The maintenance is not working when you're using varnish and you're not logged in?

**Comment by highacid** [comment]
I am logged in, that is the point, when I see the main page, it shows me maintenance mode, if i disable only i can see main page.

**Comment by georgejipa** [comment]
If you disable varnish everything is fine? Please post here your varnish configuration. Also, do you use WP Multisite?

**Comment by highacid** [comment]
No multisite here, simple Apache, Nginx proxy Varnish and Opcache. No cache plugins.

**Comment by highacid** [comment]
```
include "/usr/local/varnish/etc/varnish/cpanel.backend.vcl";
include "/usr/local/varnish/etc/varnish/vhost.vcl";

sub vcl_recv {

  set req.backend = default;
  include "/usr/local/varnish/etc/varnish/acl.vcl";
  include "/usr/local/varnish/etc/varnish/vhost.exclude.vcl";
  set req.grace = 5m;
  
  # Handle IPv6
  if (req.http.Host ~ "^ipv6.*") {
    set req.http.host = regsub(req.http.host, "^ipv6\.(.*)","www\.\1");
  }

  # Sanitise X-Forwarded-For...
  remove req.http.X-Forwarded-For;
  set req.http.X-Forwarded-For = client.ip;
  include "/usr/local/varnish/etc/varnish/cpanel.url.vcl"; 
  # Remove has_js and Google Analytics cookies.
  set req.http.Cookie = regsuball(req.http.Cookie, "(^|;\s*)(__[a-z]+|has_js)=[^;]*", "");
  # Remove a ";" prefix, if present.
  set req.http.Cookie = regsub(req.http.Cookie, "^;\s*", ""); 
  # Normalize the Accept-Encoding header
  
  if (req.http.Accept-Encoding) {
    if (req.url ~ "\.(jpg|jpeg|png|gif|gz|tgz|bz2|tbz|mp3|ogg|swf|flv|pdf|ico)$") {
      # No point in compressing these
      remove req.http.Accept-Encoding;
    } 
    elsif (req.http.Accept-Encoding ~ "gzip") {
      set req.http.Accept-Encoding = "gzip";
    } 
    elsif (req.http.Accept-Encoding ~ "deflate") {
      set req.http.Accept-Encoding = "deflate";
    } 
    else {
      # unknown algorithm
      remove req.http.Accept-Encoding;
    }
  }
  
  include "/usr/local/varnish/etc/varnish/url.exclude.vcl"; 
 
# Unset Cookies except for WordPress admin and WooCommerce pages 
if (!(req.url ~ "(wp-login|wp-admin|cart|warenkorb|my-account/*|mein-konto/*|checkout|kasse|addons|logout|lost-password|product/*)")) { 
unset req.http.cookie; 
} 
# Pass through the WooCommerce dynamic pages 
if (req.url ~ "^/(cart|warenkorb|my-account/*|mein-konto/*|checkout|kasse|addons|logout|lost-password|product/*|produkt/*)") { 
return (pass); 
} 
# Pass through the WooCommerce add to cart 
if (req.url ~ "\?add-to-cart=" ) { 
return (pass); 
} 
# Block access to php admin pages via website 
if (req.url ~ "^/phpmyadmin/.*$" || req.url ~ "^/phppgadmin/.*$" || req.url ~ "^/server-status.*$") { 
error 403 "For security reasons, this URL is only accesible using localhost (127.0.0.1) as the hostname"; 
} 
#

 
  # Ignore empty cookies
  if (req.http.Cookie ~ "^\s*$") {
    remove req.http.Cookie;
  }
  
  if (req.request == "BAN") {
    if (!client.ip ~ acl127_0_0_1) {
      error 405 "Not allowed.";
    }
    # This option is to clear any cached object contain ing the req.url
    ban("req.url ~ "+req.url);

    # This option is to clear any cached object matches the exact req.url
    # ban("req.url == "+req.url);

    # This option is to clear any cached object containing the req.url 
    # AND matching the hostname.
    # ban("req.url ~ "+req.url+" && req.http.host == "+req.http.host);

    error 200 "Cache Cleared Successfully.";
  }    

  if (req.request != "GET" &&
       req.request != "HEAD" &&
       req.request != "POST" &&
       req.request != "PUT" &&
       req.request != "PURGE" &&
       req.request != "DELETE" ) 
  {
    return (pipe);    
  }

  if (req.request != "GET" && req.request != "HEAD") {
    /* We only deal with GET and HEAD by default, the rest get passed direct to backend */
    return (pass);
  }
  
  if (req.http.Cookie ~ "^\s*$") {
    unset req.http.Cookie;
  }
  
  if (req.http.Authorization || req.http.Cookie) {
    return (pass);
  }
  
  # don't cache ajax requests
  if (req.http.X-Requested-With == "XMLHttpRequest" || 
	req.url ~ "nocache" || 
	req.url ~ "(control.php|wp-comments-post.php|wp-login.php|register.php)") 
  {
    return (pass);
  } 
 
  set req.url = regsub(req.url, "\.js\?.*", ".js");
  set req.url = regsub(req.url, "\.css\?.*", ".css");
  set req.url = regsub(req.url, "\.jpg\?.*", ".jpg");
  set req.url = regsub(req.url, "\.gif\?.*", ".gif");
  set req.url = regsub(req.url, "\.swf\?.*", ".swf");
  set req.url = regsub(req.url, "\.xml\?.*", ".xml");

  # Cache things with these extensions
  if (req.url ~ "\.(js|css|jpg|jpeg|png|gif|gz|tgz|bz2|tbz|mp3|ogg|swf|pdf)$" && ! (req.url ~ "\.(php)") ) {
    unset req.http.Cookie;
    return (lookup);
  }
  
  if (! req.backend.healthy) {
    set req.grace = 1h;
  }
  
  return (lookup);
}


sub vcl_fetch {

  set beresp.ttl = 45s;
  set beresp.http.Server = " - ApacheBooster by http://www.ndimensionz.com";
  set beresp.do_gzip = true;
  set beresp.do_gunzip = false;
  set beresp.do_stream = false;
  set beresp.do_esi = false;
  set beresp.grace = 5m;
  unset beresp.http.expires;

  if (req.url ~ "\.(js|css|jpg|jpeg|png|gif|gz|tgz|bz2|tbz|mp3|ogg|swf|pdf|ico)$" && ! (req.url ~ "\.(php)") ) {
    unset beresp.http.set-cookie;
    include  "/usr/local/varnish/etc/varnish/static_file.vcl";
  }
  else {
    include  "/usr/local/varnish/etc/varnish/dynamic_file.vcl";
  }

  if (beresp.status == 503 || beresp.status == 500) {
    set beresp.http.X-Cacheable = "NO: beresp.status";
    set beresp.http.X-Cacheable-status = beresp.status;
    return (hit_for_pass);
  }

  if (beresp.status == 404) {
    set beresp.http.magicmarker = "1";
    set beresp.http.X-Cacheable = "YES";
    set beresp.ttl = 20s;
    return (deliver);
  }

  set beresp.http.magicmarker = "1";
  set beresp.http.X-Cacheable = "YES";

# Unset Cookies except for WordPress admin and WooCommerce pages 
if ( (!(req.url ~ "(wp-(login|admin)|login|cart|warenkorb|my-account/*|mein-konto/*|checkout|kasse|addons|logout|lost-password|product/*|produkt/*)")) || (req.request == "GET") ) { 
unset beresp.http.set-cookie; 
} 
#
}

sub vcl_deliver {

  if ( obj.hits == 0 ) {
    set req.http.X-Stats-HitMiss = "miss";
  } 
  else {
    set req.http.X-Stats-HitMiss = "hit";
  }
}

```

**Comment by georgejipa** [comment]
If you disable varnish everything is fine? I'll make some tests on my dev server.

**Comment by highacid** [comment]
Think is I cannot really disable it, it will off my sites cause is tight with nginx in such a way binded.

**Comment by georgejipa** [comment]
Based on your config, varnish is disabled (no cache) when maintenance mode is activated. So I don't think it has something to do with varnish. Maybe there is a conflict with another plugin.

```
  if (beresp.status == 503 || beresp.status == 500) {
    set beresp.http.X-Cacheable = "NO: beresp.status";
    set beresp.http.X-Cacheable-status = beresp.status;
    return (hit_for_pass);
  }
```

What version of WP Maintenance Mode is installed? The latest?

**Comment by highacid** [comment]
yes i use latest no plugins were installed lately...

**Comment by georgejipa** [comment]
Unfortunately, I can't replicate this issue on my development machine. There is a similar issue with WP Multisite (which is solved in 2.0.8 and will be released in a few days), but works fine on simple WP install (like you said you have).
